src/AspNetCoreModuleV2/CommonLib/Environment.cpp

```diff
@@ -56,5 +56,7 @@ Environment::GetEnvironmentVariableValue(const std::wstring & str)
         }
     } while (expandedStr.size() != requestedSize + 1);
 
+    expandedStr.resize(requestedSize);
```

Comment:
Justification for this change? Seems like something we should add a test for.

Reply:
I didn't work correctly leaving `\0` in string contents.

Reply:
Added test.